custom/crediblex/portfolio/loc/src/main/java/com/crediblex/fineract/portfolio/loc/service/LineOfCreditDataValidator.java

```diff
@@ -361,6 +388,84 @@ private void validateNewFields(JsonElement element, DataValidatorBuilder baseDat
         }
     }
 
+    /**
+     * Validates that a line of credit can be deactivated.
+     * Disallows deactivation of LOC accounts with active loan accounts linked to them.
+     *
+     * @param lineOfCreditId the ID of the line of credit to validate for deactivation
+     */
+    public void validateForDeactivation(Long lineOfCreditId) {
+        if (lineOfCreditId == null) {
+            throw new IllegalArgumentException("Line of credit ID cannot be null");
+        }
+
+        final List<ApiParameterError> dataValidationErrors = new ArrayList<>();
+        final DataValidatorBuilder baseDataValidator = new DataValidatorBuilder(dataValidationErrors).resource("line.of.credit");
+
+        // Check if there are any active loan accounts linked to this line of credit
+        String sql = "SELECT COUNT(*) FROM m_loan l WHERE l.line_of_credit_id = ? AND l.loan_status_id = 300";
+        
+        try {
+            Integer activeLoanCount = jdbcTemplate.queryForObject(sql, Integer.class, lineOfCreditId);
+            
+            if (activeLoanCount != null && activeLoanCount > 0) {
+                baseDataValidator.reset().parameter("lineOfCreditId").value(lineOfCreditId)
+                    .failWithCode("line.of.credit.cannot.deactivate.with.active.loans", 
+                        "Cannot deactivate line of credit with ID " + lineOfCreditId + 
+                        " because it has " + activeLoanCount + " active loan account(s) linked to it.");
+            }
+        } catch (Exception e) {
+            baseDataValidator.reset().parameter("lineOfCreditId").value(lineOfCreditId)
+                .failWithCode("line.of.credit.deactivation.validation.error", 
```

Comment:
Dont use try catch. I only expect one try catch around the jdbc query and it should be a specific exceptionfor when there no object returned.